evolver/app/routers/hardware.py

```diff
@@ -139,7 +140,9 @@ def get_calibrator_state(hardware_name: str, request: Request):
     hardware_instance = get_hardware_instance(request, hardware_name)
     if not (calibrator := hardware_instance.calibrator):
         raise CalibratorNotFoundError
-    calibration_procedure = calibrator.calibration_procedure
+    if not (calibration_procedure := calibrator.calibration_procedure):
+        raise CalibrationProcedureNotFoundError
```

Comment:
should we also add this to the same line in `def dispatch_calibrator_action` above?